redwood-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyList.kt

```diff
@@ -121,7 +106,8 @@ internal open class UIViewLazyList : LazyList<UIView>, ChangeListener {
         layout: UICollectionViewLayout,
         sizeForItemAtIndexPath: NSIndexPath,
       ): CValue<CGSize> {
-        val itemSize = items.itemForGlobalIndex(sizeForItemAtIndexPath.item.toInt()).size
+        val widget = items.itemForGlobalIndex(sizeForItemAtIndexPath.item.toInt())
+        val itemSize = widget.value.sizeThatFits(collectionView.frame().useContents { size.readValue() })
```

Comment:
Is this performant? The whole reason I added the `ViewPortItem` class in PR #1328 was so that we could pre-compute the `sizeThatFits` on `insert` (called once per widget), rather than here (called repeatedly on scroll).

Reply:
I think we should assume a slight degradation in performance because of #1341, where we now have more widgets to layout and size. This wouldn't account for such a large jump in performance though. Running EmojiSearch and removing the usage of `Row` in each item shows _significant_ performance improvements. Currently looking into that.

Regarding your second paragraph, I 100% agree. I've gotten used to Redwood being in development, that a "one step backwards, two steps forward" approach on `trunk` is no longer permissible, due to the broader impact that Redwood has internally. Will definitely keep that in mind in the future!

For the current breakage, I'll continue to try to fix it. If it takes longer, or we need an internal release, I can make the necessary rollbacks, and continue development on a separate branch.

Reply:
I think this is the underlying issue (it is to do with `Row`/`Column`): https://github.com/cashapp/redwood/issues/1205

If you comment out the following line, performance drastically improves. https://github.com/cashapp/redwood/blob/3f256fa271ea6d8913838cad019f2e0bf6cae212/redwood-yoga/src/commonMain/kotlin/app/cash/redwood/yoga/Node.kt#L98

Reply:
Thanks for continuing to dig into this!